src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -34,12 +34,17 @@ public function process(ContainerBuilder $container)
         $types = array();
 
         foreach ($container->findTaggedServiceIds('form.type') as $serviceId => $tag) {
-            $alias = isset($tag[0]['alias'])
-                ? $tag[0]['alias']
-                : $serviceId;
+            // The following if-else block is deprecated and can be removed
+            // in Symfony 3.0
+            if (isset($tag[0]['alias'])) {
+                $types[$tag[0]['alias']] = $serviceId;
+            } else {
+                $types[$serviceId] = $serviceId;
```

Comment:
Service IDs are used throughout symfony to reference services, so it feels natural to me to also possibly let them be used to create forms. I don't really see why you said "it doesn't make sense". So I think we should keep that feature (although I haven't used it this way, but it feels consistent).